packages/trackerless-network/protos/NetworkRpc.proto

```diff
@@ -47,48 +43,44 @@ enum StreamMessageType {
   GROUP_KEY_RESPONSE = 2;
 }
 
-enum EncryptionType {
-  NONE = 0;
-  RSA = 1;
-  AES = 2;
+enum ContentType {
+  JSON = 0;
+  BINARY = 1;
 }
 
-message EncryptedGroupKey {
-  string groupKeyId = 1;
-  string encryptedGroupKeyHex = 2;
-  optional string serialized = 3;
+enum EncryptionType {
+  NONE = 0;
+  AES = 1;
 }
 
 message StreamMessage {
   StreamMessageType messageType = 1;
-  optional EncryptionType encryptionType = 2;
-  bytes content = 3;
-  string signature = 4;
-  MessageRef messageRef = 5;
-  optional MessageRef previousMessageRef = 6;
-  optional string groupKeyId = 7;
-  optional EncryptedGroupKey newGroupKey = 8;
+  ContentType contentType = 2;
+  EncryptionType encryptionType = 3;
+  bytes content = 4;
+  bytes signature = 5;
+  MessageRef messageRef = 6;
+  optional MessageRef previousMessageRef = 7;
+  optional string groupKeyId = 8;
+  optional EncryptedGroupKey newGroupKey = 9;
```

Comment:
We could improve the naming here for clarity (I know this mimics the existing `StreamMessage` structure).

Should it be
```
optional string requestedGroupKeyId = 8
```
to differentiate between what fields `groupKeyId` and `newGroupKey.groupKeyId`  refer to?

@teogeb do you remember the difference between these two fields?

Reply:
The `groupKeyId` id the group key used to encrypt this message. 

The `newGroupKey` is a new key which is delivered here for future use. In practice it is used in key rotation so that all subscribers get the key automatically. Usually we store it locally and use it to decode the next message (as the `groupKeyId` of that message will be the value of this message's `newGroupKey.id`)

Reply:
Aahh never mind my comment then.

Reply:
Let's keep it as is then.